src/s3filesystem.cpp

```diff
@@ -208,7 +208,7 @@ StatusCode S3FileSystem::isDirectory(const std::string& path, bool* is_dir) {
 
     auto head_bucket_outcome = client_.HeadBucket(head_request);
     if (!head_bucket_outcome.IsSuccess()) {
-        SPDLOG_LOGGER_ERROR(s3_logger, "Couldn't get MetaData for bucket with name {}", bucket);
+        SPDLOG_LOGGER_ERROR(s3_logger, "Invalid or missing S3 credentials, or bucket does not exist - {}", bucket);
```

Comment:
why two error messages for the same scenario?

Reply:
Because the is no API to check which of the errors occurred. It is either 3 of those in this case.

Reply:
I meant the message below. The error message head_bucket_outcome.GetError().GetMessage() could be in the same line like above.